tasks/Task.cpp

```diff
@@ -16,17 +16,69 @@ Task::~Task()
 {
 }
 
-static RigidBodyState convertToRBS(PingStatus const& data)
+static RigidBodyState convertToZWithOrientationRBS(Status const& data)
+{
+    RigidBodyState rbs;
+    rbs.time = data.timestamp;
+    rbs.position = Eigen::Vector3d(0, 0, data.environment.depth / 10.);
```

Comment:
This is used to report a Pose, (0,0) is a valid coordinate but in this case we dont have any valid data for X and Y. You should be passing a Vector3d as `(NaN, NaN, data.environment.depth / 10.0)`